hbase-server/src/main/java/org/apache/hadoop/hbase/io/FileLink.java

```diff
@@ -269,6 +269,7 @@ private FSDataInputStream tryOpen() throws IOException {
             }
           }
           currentPath = path;
+          fileLink.setReaderPath(path);
```

Comment:
I think it's odd that doing `new FileLinkInputStream(fs, fileLink)` modifies my passed in FileLink.

Maybe not the end of the world, but seems better to keep immutable. Is there any other way to do what you're trying to do without mutating the passed in link?

Reply:
We can also reuse the `currentPath` of FileLinkInputStream here.